src/Bridge/Doctrine/EventListener/PurgeHttpCacheListener.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+declare(strict_types=1);
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ApiPlatform\Core\Bridge\Doctrine\EventListener;
+
+use ApiPlatform\Core\Api\IriConverterInterface;
+use ApiPlatform\Core\Exception\InvalidArgumentException;
+use Doctrine\ORM\Event\OnFlushEventArgs;
+use Psr\Container\ContainerInterface;
+
+/**
+ * Stores IRIs of entity to purge from the proxy cache.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ *
+ * @experimental
+ */
+final class PurgeHttpCacheListener
+{
+    private $container;
+
+    /**
+     * @param ContainerInterface $container the container is injected because of a circular reference
+     */
+    public function __construct($container)
+    {
+        $this->container = $container;
+    }
+
+    public function onFlush(OnFlushEventArgs $eventArgs)
+    {
+        if (!$request = $this->container->get('request_stack')->getCurrentRequest()) {
+            return;
+        }
+
+        $resources = $request->attributes->get('_resources', []);
```

Comment:
Yes I'll try to find a better solution but I'm not sure there is one. Anyway its called `purge HTTP cache`, it's not a big deal to couple it with the HTTP request.